### ISSUE TITLE
About Barrier Setting

### ISSUE THREAD

**Issue by datutu-L** [zzzzzyh111/NonExpert-Guided-Visual-UAV-Navigation-Gazebo#3]
Hello author, thank you for such a great program! :blush:
I'm a beginner using this program for the first time, and I have a question for you, the gazebo that opens when I run `roslaunch hector_quadrotor_gazebo quadrotor_empty_world.launch` only has a drone in it, do I need to add my own obstacles before I can train with 'python main.py' ?

**Comment by zzzzzyh111** [comment]
Yes. By checking the ```quadrotor_empty_world.launch file```, you can identify the specific world file being loaded. Then, you can either add obstacles directly within the simulation and save the scene as a world file, or modify the world file directly to include the obstacles.

**Comment by datutu-L** [comment]
Thank you very much for your prompt reply. 😊

**Comment by datutu-L** [comment]
I'm sorry, but I've run into another problem that I need to bother asking you about. I am getting the following error when I run it, have you encountered this before.
Looking forward to your reply again

![image](https://github.com/user-attachments/assets/4b92656c-ab84-4370-8e54-79edaecd3a78)

**Comment by datutu-L** [comment]
The problem seems to be in this file [env.py](https://github.com/zzzzzyh111/NonExpert-Guided-Visual-UAV-Navigation-Gazebo/blob/main/env.py) which converts Ros image to cv2 image. Is there any additional settings needed other than adding obstacles?
Running `roslaunch hector_quadrotor_gazebo quadrotor_empty_world.launch` doesn't seem to turn on images, at least there is no message in my topic list for the item `/front_cam/camera/image`, all resulting in the subscribed image being NonType.
![image](https://github.com/user-attachments/assets/1db1e188-3bd6-47f4-97a1-b0f74a2cd5ff)

**Comment by datutu-L** [comment]
Do I need any other additional action to turn on the camera so that gazebo can post the `/front_cam/camera/image message`?
I'm very sorry to bother you again and look forward to your reply! :dizzy:

**Comment by zzzzzyh111** [comment]
I don't see any drone related topic in rostopic, what drone model are you using and can it be loaded properly by the launch file?

**Comment by datutu-L** [comment]
I configured [hector-quadrotor-noetic](https://github.com/RAFALAMAO/hector-quadrotor-noetic) by following this link you provided, and here is what running `roslaunch hector_quadrotor_ gazebo quadrotor_empty_world.launch` to open the interface and the rostopic list.
![image](https://github.com/user-attachments/assets/951056be-7f34-440a-8eaa-186c4e5ba75e)

**Comment by zzzzzyh111** [comment]
You're loading the wrong launch file, this is simply a quadrotor file, look inside the folder and there should be a launch file like quadrotor_with_camera, load that to get it right.

**Comment by olivinelulu** [comment]
> 抱歉，我又遇到了一个问题，需要麻烦您咨询一下。我运行程序时出现了以下错误，您以前遇到过这种情况吗？ 期待您的再次回复。
> 
> ![image](https://github.com/user-attachments/assets/4b92656c-ab84-4370-8e54-79edaecd3a78)

How was this problem solved?I encountered the same issue.Thank you very much!

**Comment by zzzzzyh111** [comment]
Please see https://github.com/zzzzzyh111/NonExpert-Guided-Visual-UAV-Navigation-Gazebo/issues/4#issuecomment-2872326526
